lib/Document.php

```diff
@@ -221,14 +211,14 @@ public function createProperty($name, $value = null, array $parameters = null, $
         if (is_null($class)) {
             // If a VALUE parameter is supplied, we should use that.
             if (isset($parameters['VALUE'])) {
-                $class = $this->getClassNameForPropertyValue($parameters['VALUE']);
-                if (is_null($class)) {
-                    if (in_array(strtoupper($parameters['VALUE']), $this->allowedIllegalValues, true)) {
-                        unset($parameters['VALUE']);
-                        $class = $this->getClassNameForPropertyName($name);
-                    } else {
-                        throw new InvalidDataException('Unsupported VALUE parameter for '.$name.' property. You supplied "'.$parameters['VALUE'].'"');
-                    }
+                try {
+                    $class = $this->getClassNameForPropertyValue($parameters['VALUE']);
+                } catch (\Throwable $t) {
+                    $class = null;
+                }
+                if (is_null($class)) { // VALUE is malformed or illegal, drop it
+                    unset($parameters['VALUE']);
```

Comment:
Does leaving the value hurt us further down the line?

Reply:
good question, let me try

Reply:
so sabre vobject were just throwing exception if they don't find it in their mapping https://github.com/sabre-io/vobject/blob/master/lib/Document.php#L196

and technically the list for valid VALUE data are defined in the RFC https://datatracker.ietf.org/doc/html/rfc5545#section-3.2.20

so the previous fix from us is to allow VALUE=ERROR but still drop it from the parsed ICS

dropping it here for all invalid values are safer imho, I can't really tell where the invalid value will break down the line, if we really want to be fully RFC compliant, we need to somehow preserve the VALUE if they fall into any of the extra categories like x-name, experimental type, etc.